### PR TITLE
Changing Xamarin.Forms to support offline sync

### PR DESCRIPTION
**Pull request by fabiocav**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@fabiocav__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by lindydonna** [comment]
Need to fix CLA automation; CLA is not required

File: client/xamarin.forms/iOS/ZUMOAPPNAME.iOS.tt [hunk 1]
```diff
@@ -65,16 +65,42 @@
...
+      <SpecificVersion>False</SpecificVersion>
       <HintPath>..\packages\WindowsAzure.MobileServices.<#= ManagedSDKVersion #>\lib\Xamarin.iOS10\Microsoft.WindowsAzure.Mobile.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.WindowsAzure.Mobile.Ext">
+      <SpecificVersion>False</SpecificVersion>
       <HintPath>..\packages\WindowsAzure.MobileServices.<#= ManagedSDKVersion #>\lib\Xamarin.iOS10\Microsoft.WindowsAzure.Mobile.Ext.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.WindowsAzure.Mobile.SQLiteStore, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\packages\WindowsAzure.MobileServices.SQLiteStore.2.0.0-beta\lib\portable-win+net45+wp8+wpa81+monotouch+monoandroid\Microsoft.WindowsAzure.Mobile.SQLiteStore.dll</HintPath>
```
Comment (phvannor) [inline client/xamarin.forms/iOS/ZUMOAPPNAME.iOS.tt]: versions in teh .tt files need replaced with the build vars

File: client/xamarin.forms/iOS/packages.tt [hunk 2]
```diff
@@ -4,10 +4,12 @@
 <#@parameter type="System.String" name="StoreSDKVersion"#>
 <?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Bcl" version="1.1.10" targetFramework="xamarinios10" />
-  <package id="Microsoft.Bcl.Build" version="1.0.21" targetFramework="xamarinios10" />
-  <package id="Microsoft.Net.Http" version="2.2.29" targetFramework="xamarinios10" />
-  <package id="Newtonsoft.Json" version="6.0.8" targetFramework="xamarinios10" />
+  <package id="Microsoft.Bcl" version="1.1.9" targetFramework="xamarinios10" />
```
Comment (phvannor) [inline client/xamarin.forms/iOS/packages.tt:7]: this is a downgrade?  Why go back to 1.1.9?

File: client/xamarin.forms/iOS/packages.tt [hunk 3]
```diff
@@ -4,10 +4,12 @@
...
+  <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="xamarinios10" />
+  <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="xamarinios10" />
+  <package id="Newtonsoft.Json" version="6.0.4" targetFramework="xamarinios10" />
+  <package id="SQLitePCL" version="3.8.7.2" targetFramework="xamarinios10" />
   <package id="WindowsAzure.MobileServices" version="<#= ManagedSDKVersion #>" targetFramework="xamarinios10" />
+  <package id="WindowsAzure.MobileServices.SQLiteStore" version="<#= ManagedSDKVersion #>" targetFramework="xamarinios10" />
```
Comment (phvannor) [inline client/xamarin.forms/iOS/packages.tt]: shouldn't the SQLiteStore have its own version?  It shouldn't always equal the ManagedSDKVersion.

File: client/xamarin.forms/ZUMOAPPNAME/TodoItemManager.cs [hunk 4]
```diff
@@ -22,71 +24,81 @@ public TodoItemManager()
...
-	}
+            //Initializes the SyncContext using the default IMobileServiceSyncHandler.
+            this.client.SyncContext.InitializeAsync(store);
+
+            this.todoTable = client.GetSyncTable<TodoItem>();
+        }
+
+        public async Task SyncAsync()
+        {
+            await this.todoTable.PullAsync("todoItems", this.todoTable.CreateQuery());
```
Comment (phvannor) [inline client/xamarin.forms/ZUMOAPPNAME/TodoItemManager.cs]: as I mentioned, it'd be nice if this handled some error conditions gracefully.

File: client/xamarin.forms/iOS/packages.tt (hunk 3 above)
Comment (fabiocav) [inline client/xamarin.forms/iOS/packages.tt]: Updated all SQLiteStore references with the correct property name (25492a8e411f895a6291f75b1dbc0645678f43ad).

File: client/xamarin.forms/iOS/packages.tt (hunk 2 above)
Comment (fabiocav) [inline client/xamarin.forms/iOS/packages.tt:7]: This is the minimum required version for this project. If we want to change it back to 1.1.10 we can, but it isn't required.

**Comment by phvannor** [comment]
Part of me would like this done on Master and pulled over to MobileApp.  Any reason its not applied to both?

File: client/xamarin.forms/ZUMOAPPNAME/TodoItemManager.cs (hunk 4 above)
Comment (fabiocav) [inline client/xamarin.forms/ZUMOAPPNAME/TodoItemManager.cs]: Added some basic error handling to bring it to parity with what we have for iOS.

**Pull request closed without merging**